XAlloc.h

```diff
@@ -35,4 +35,10 @@ void* xRealloc(void* ptr, size_t size);
 
 char* xStrdup_(const char* str) ATTR_NONNULL;
 
+static inline void* xFree(void* ptr) {
```

Comment:
I don't quite overall like this function, but see where it comes from. The classical idiom is writing `free(x); x = NULL;`, thus hiding the `NULL` in the function call is not the best style IMHO.